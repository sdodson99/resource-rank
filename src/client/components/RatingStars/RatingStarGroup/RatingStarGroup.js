import React from 'react';
import PropTypes from 'prop-types';
import styles from './RatingStarGroup.module.css';

const RatingStarGroup = () => (
  <div className={styles.RatingStarGroup} data-testid="RatingStarGroup">
    RatingStarGroup Component
  </div>
);

RatingStarGroup.propTypes = {};

RatingStarGroup.defaultProps = {};

export default RatingStarGroup;
