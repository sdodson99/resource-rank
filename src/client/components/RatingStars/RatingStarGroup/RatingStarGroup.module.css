.RatingStarGroup {
  margin-left: -3px;
  
  @apply flex;
}
