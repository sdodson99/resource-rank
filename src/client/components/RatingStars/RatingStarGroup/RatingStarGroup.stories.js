import React from 'react';
import RatingStarGroup from './RatingStarGroup';

export default {
  component: RatingStarGroup,
  title: 'Components/RatingStars/RatingStarGroup',
  args: {
    starSize: 50,
  },
};

const Template = (args) => <RatingStarGroup {...args} />;

export const Default = Template.bind({});
Default.args = {};

export const Size = Template.bind({});
Size.args = {
  starSize: 100,
};

export const MaxRating = Template.bind({});
MaxRating.args = {
  maxRating: 12,
};

export const EmptyRating = Template.bind({});
EmptyRating.args = {
  rating: 0,
};

export const HalfRating = Template.bind({});
HalfRating.args = {
  rating: 2.5,
};

export const FullRating = Template.bind({});
FullRating.args = {
  rating: 5,
};
