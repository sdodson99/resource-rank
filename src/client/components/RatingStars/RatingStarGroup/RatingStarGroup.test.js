import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import RatingStarGroup from './RatingStarGroup';
import { createRenderer } from 'react-test-renderer/shallow';

describe('<RatingStarGroup />', () => {
  test('it should mount', () => {
    render(<RatingStarGroup />);

    const ratingStarGroup = screen.getByTestId('RatingStarGroup');

    expect(ratingStarGroup).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const component = createRenderer().render(
      <RatingStarGroup starSize={50} maxRating={5} />
    );

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with empty rating', () => {
    const component = createRenderer().render(<RatingStarGroup rating={0} />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with half rating', () => {
    const component = createRenderer().render(<RatingStarGroup rating={2.5} />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with full rating', () => {
    const component = createRenderer().render(<RatingStarGroup rating={5} />);

    expect(component).toMatchSnapshot();
  });

  it('should execute onStarClick callback for star index when star clicked', () => {
    const mockOnStarClick = jest.fn();
    const clickedStarIndex = 2;
    render(<RatingStarGroup onStarClick={mockOnStarClick} />);
    const ratingStar = screen.getAllByTestId('RatingStarGroup_RatingStar')[
      clickedStarIndex
    ];

    ratingStar.click();

    expect(mockOnStarClick).toBeCalledWith(clickedStarIndex);
  });
});
