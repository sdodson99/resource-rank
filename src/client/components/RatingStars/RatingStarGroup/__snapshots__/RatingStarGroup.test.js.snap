// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RatingStarGroup /> should render correctly 1`] = `
<div
  className="RatingStarGroup"
  data-testid="RatingStarGroup"
>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={0}
      size={50}
    />
  </div>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={-1}
      size={50}
    />
  </div>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={-2}
      size={50}
    />
  </div>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={-3}
      size={50}
    />
  </div>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={-4}
      size={50}
    />
  </div>
</div>
`;

exports[`<RatingStarGroup /> should render correctly with empty rating 1`] = `
<div
  className="RatingStarGroup"
  data-testid="RatingStarGroup"
>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={0}
      size={20}
    />
  </div>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={-1}
      size={20}
    />
  </div>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={-2}
      size={20}
    />
  </div>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={-3}
      size={20}
    />
  </div>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={-4}
      size={20}
    />
  </div>
</div>
`;

exports[`<RatingStarGroup /> should render correctly with full rating 1`] = `
<div
  className="RatingStarGroup"
  data-testid="RatingStarGroup"
>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={5}
      size={20}
    />
  </div>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={4}
      size={20}
    />
  </div>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={3}
      size={20}
    />
  </div>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={2}
      size={20}
    />
  </div>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={1}
      size={20}
    />
  </div>
</div>
`;

exports[`<RatingStarGroup /> should render correctly with half rating 1`] = `
<div
  className="RatingStarGroup"
  data-testid="RatingStarGroup"
>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={2.5}
      size={20}
    />
  </div>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={1.5}
      size={20}
    />
  </div>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={0.5}
      size={20}
    />
  </div>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={-0.5}
      size={20}
    />
  </div>
  <div
    className="mr-1"
    data-testid="RatingStarGroup_RatingStar"
    onClick={[Function]}
  >
    <RatingStar
      color="orange"
      fill={-1.5}
      size={20}
    />
  </div>
</div>
`;
